### DISCUSSION TITLE
Conflict between packages

### DISCUSSION THREAD

**Discussion by 01fiore** [mocaccinoOS/mocaccino#147] [General] [answered]
hello, I have a problem with the systema update:

$ sudo luet upgrade
Mot de passe :
 INFO  Luet 0.32.5-gd5d21653d7313ff396c0511af402834c89974ee3 2022-06-07
 INFO  Copyright (C) 2019-2022 Ettore Di Giacinto
       This program comes with ABSOLUTELY NO WARRANTY.
       This is free software, and you are welcome to redistribute it under certain conditions.
       For documentation, visit https://luet.io.
 INFO   :::> Upgrade
 INFO   🤔  Computing upgrade, please hang tight... 💤
 INFO   ⚡  Proposed version changes to the system:

Old version                           | New version                           | License
layers/kde-common-6+19                | layers/kde-common-6+22                |
layers/X-1.3+38                       | layers/X-1.3+39                       |
kernel/mocaccino-lts-full-6.6.51      | kernel/mocaccino-lts-full-6.6.54      |
layers/gnome-common-1.2+83            | layers/gnome-common-1.2+84            |
layers/system-x-0.20240923            | layers/system-x-0.20241005            |
kernel/mocaccino-lts-modules-6.6.51   | kernel/mocaccino-lts-modules-6.6.54   |
apps/vlc-3.0.20-r11+15                | apps/vlc-3.0.21                       | LGPL-2.1 GPL-2
system/gcc-base-13.3.1+1              | system/gcc-base-13.3.1+2              |
kernel/mocaccino-initramfs-6.11.1     | kernel/mocaccino-initramfs-6.11.2     |
apps/flatpak-1.14.8+11                | apps/flatpak-1.14.8+12                | LGPL-2.1+
apps/gparted-1.6.0-r1+2               | apps/gparted-1.6.0-r1+3               | GPL-2+ FDL-1.2+
layers/plasma-6.1.5                   | layers/plasma-6.1.5+1                 | GPL-2
apps/steam-1.0.0.79+15                | apps/steam-1.0.0.79+16                | ValveSteamLicense MIT BSD
layers/kde-apps-minimal-24.05.2-r1+12 | layers/kde-apps-minimal-24.05.2-r1+13 |
kernel/mocaccino-lts-initramfs-6.6.51 | kernel/mocaccino-lts-initramfs-6.6.54 |
kernel/mocaccino-lts-minimal-6.6.51   | kernel/mocaccino-lts-minimal-6.6.54   |
apps/thunderbird-115.15.0+2           | apps/thunderbird-128.3.0              | MPL-2.0 GPL-2 LGPL-2.1
layers/qt-5.15.14+31                  | layers/qt-5.15.14+32                  |
apps/firefox-115.15.0+2               | apps/firefox-128.3.0                  | MPL-2.0 GPL-2 LGPL-2.1
layers/sys-fs-1.1+46                  | layers/sys-fs-1.1+47                  |

 INFO   By going forward, you are also accepting the licenses of the packages that you are going to install in your system.
 INFO   Do you want to continue with this operation? [y/N]:
y
 INFO   Checking for file conflicts..
  ERROR    Error: file conflict found: file conflict between packages to be installed

**Comment by msdobrescu** [comment]
Let's try to assess the duplicates by running the following command as root/su:

`luet search -o json | jq '{ packages: [.packages[] | select( .category != "kernel" and ( .name | contains("nvidia") | not ) )] }' | luet filter`

There are packages that are mutually exclusive (like kernels or nvidia drivers sue to the fact there are several version available - those are excluded above).

There are packages that may have indeed common files, but these are related to cases when desktops and apps have such files. Those are usually not common to happen because the desktop in question and that app should not work together, but there are rare cases when multiple desktops are installed at the same time. This should not happen between latest Gnome and Plasma.

**Comment by 01fiore** [comment]
I reinstalled all the packages individually without problem, before giving the reboot I do an upgrade where he told me that everything was ok, to reboot black screen.

As Macacinno no more shares the commands of Sabayon and Gentoo, I can not play with it, I will try later. Greetings.

**Comment by msdobrescu** [comment]
I'd boot to command line and run a `luet oscheck --reinstall` command`, this should look for missing files from the installed packages. If there is something to reinstall, it means that some  files are moved between packages for dependency reasons and `luet` may choose an order of operations that won't bring those files back (would upgrade them first with one package that owns them in the newer version, then remove the package that owned them in the first place, the old version). As a good practice, it's advised to run this command after upgrade every time.
Migrating to Plasma 6 (QT6 + KDE/KF6) was pretty complicated, so there were many layers transforms, including some layers dropping at some step. Those must be cleaned up completely.

**Comment by joostruis** [comment, marked as answer]
kernel/mocaccino-initramfs-6.11.1 | kernel/mocaccino-initramfs-6.11.2 |

You messed around with kernels and now you have part LTS kernel and part mainline kernel.

The issue:
https://github.com/mocaccinoOS/mocaccino/issues/109

How to resolve:
`luet uninstall kernel/mocaccino-initramfs`

Then:
`luet install kernel/mocaccino-lts-initramfs`

and 

`mocaccino-dracut --rebuild-all`

**Reply by 01fiore** [reply]
Unable to access the system after update, I reinstalled the OS and updated without problem, next time I wait before putting random commands. I have price note orders if the problem reoccurs.

Ciao.

**Reply by joostruis** [reply]
You are welcome to ask advice here anytime!
